### DISCUSSION TITLE
Quick Permanent fix for Debian ARM Booting into UEFI Mapping Table

### DISCUSSION THREAD

**Discussion by UriZafrir** [utmapp/UTM#4008] [General]
This post is referring to the fix in the thread:

Ubuntu 20 AArch64 image from gallery can't boot on Apple M1 "BdsDxe: failed to load Boot0001" #2333
The problem is that debian boots into UEFI mode and when booting we have to manually activate grubaa64.efi file with the command:

`FS0:\EFI\debian\grubaa64.efi
`

I suggest a more permanent solution:
(taken from https://blog.wbcchsyn.net/1574439775.html)

First of all, let’s go to directory /EFI. Type as follows.

```
Shell> fs0:
fs0:\> cd EFI
fs0:\EFI\>
```

Then, make directory /EFI/boot and copy the files from /EFI/debian to /EFI/boot.
```
fs0:\EFI\> mkdir boot
fs0:\EFI\> cd debian
fs0:\EFI\debian\> cp grubaa64.efi ../boot/bootaa64.efi
fs0:\EFI\debian\> cp grub.cfg ../boot'
```

then

`fs0:\EFI\debian\> reset
`
Commands mkdir boot and cd debian are the same to bash. I went to directory debian to enable filename completion. (Pushing [tab] button after cp grub., then the shell complete the extension cfg if I was in the same directory to file grub.cfg.)

Command cp <src> <dst> is also same to bash. grubaa64.efi is the kickstart program Debian created. It must be rename bootaa64.efi. grub.cfg is a configuration file. It must be the same directory to the program.

**Comment by cweise04** [comment]
It works up till I type in fs0:\EFI\> cd debian then I get fs0:\EFI\> cd debian is not a directory. Am I doing something wrong? Just really trying to get Kali to work on my mac.

**Reply by UriZafrir** [reply]
You probably don't have a folder debian because you are installing kali. 
try typing the command 
ls
to see folders. 
you should probably have a folder named kali..

**Comment by RipperSK** [comment]
So for people who has struggled with UTM for a whole day trying to figure out why your ARM64 distro has stopped booting - here's a nice little "hack" you can do to see more output after grub menu disappears:

- open UTM VM edit dialog
- add serial device (with default values)
- start up your VM - it will now open 2 windows - one for VM display and one for Serial Interface
- diagnose what's preventing the boot on the serial interface window that shows the boot messages

My actual problem was that somehow the RootFS got some orphaned inodes on FS due to improper shutdown (or whatnot) and needed just a simple fsck.ext4 - but this prevented the UTM boot and I could not diagnose without seeing the console - so serial device helped.
